BLSQ/iaso
# POLIO-1335 Vaccine Stock Management backend: details view API

Adding the last APIs for Vaccine Stock Management Details Page.

Related JIRA tickets : POLIO-1335, POLIO-1334



## Changes

Refactoring and extracting from the `Serializer` the code that compute the different values about stock management in a `VaccineStockCalculator` that's used in multiple places.
Refactoring the `/` list endpoint to use it and cleanup the serializer to handle list and single case.
Adding `/summary` endpoint
Adding `/unusable_vials` endpoint
Adding tests for all this and fixing other tests.
Adding some inline documentations.

## How to test
As in the previous PR #1081 you need to have the correct data setup.
Check the tests in `plugins/polio/tests/test_vaccine_stock_management.py`
Open `/swagger/` and go to `/api/polio/vaccine/vaccine_stock/` part to test the different APIs
Visually inspect stuff in Vaccine Module Stock Management Page.
